040-SDK/150-file-attachments.mdx

```diff
@@ -7,6 +7,91 @@ slug: sdk/file-attachments
 published: true
 ---
 
+Xata provides three different ways to upload files to a file column within a database.
+
+- [Upload URLs](#upload-urls), when requested from an existing file object, provide a temporary, secure URL to upload files to. They can be used client-side and are good when dealing with large files through a client interface like a browser. This method comes at the downside of requiring extra steps across the client and server.
+- [Record APIs](#record-apis) provide the easiest way to upload files, and are the only methods that allow uploading files during the same step as creating a record. The downside is that they only support small files and will likely run over the limits of serverless functions from providers like Vercel.
```

Comment:
@snide, what do you think about calling the actual limit here instead of "small files":
... they only support files up to 20MB and ...

Reply:
Makes sense. I wavered because those things are hard to track down when they change, but if you think it's pretty constant that makes sense.